### PR TITLE
Add Fixtures

### PR DESCRIPTION
**Pull request by bitPogo**
### :tophat: What is the goal?
In preparation to migrate to Kotlin Serialization this adds Fixtures and
a bit of tooling.

### :unicorn: How is it being implemented?

it adds a primitive resource loader for strings and a test config as well as the fixtures itself.

### :thinking: DOD Checklist

- [x] My code follows the code style and naming conventions of this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have updated the documentation.
- [ ] I have updated the changelog accordingly.

### CODE REVIEW

File: crypto/src-gen/commonTest/kotlin/care/data4life/sdk/crypto/test/config/TestConfig.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+
+package care.data4life.sdk.crypto.test.config
+
+object TestConfig {
+    const val projectDir: String = "/Users/d4l000126/projects/hc-crypto-sdk-kmp/crypto"
```
Comment (wmontwe) [inline crypto/src-gen/commonTest/kotlin/care/data4life/sdk/crypto/test/config/TestConfig.kt:20]: The file should be added to gitignore and not be commited.

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/symCommonEncrypt.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  "test": "symEncrypt",
+  "input": "SGVsbG8gV29ybGQ=",
+  "output": "t7V3l3jyjXv6DWP0e9KlEiTOHXGPxBsqBCTl",
+  "iv": "uy2BwDTY4vUeBrp+",
+  "key": {
+    "t": "ck",
+    "sym": "LL85V5L0n5024C6XLGe+/mmg9F/VkUbRQycnN28S5L4=",
+    "v": 1
+  }
+}
```
Comment (wmontwe) [inline crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/symCommonEncrypt.json:11]: Linebreak on file end

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/symCommonDecrypt.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  "test": "symDecrypt",
+  "input": "t7V3l3jyjXv6DWP0e9KlEiTOHXGPxBsqBCTl",
+  "output": "SGVsbG8gV29ybGQ=",
+  "iv": "uy2BwDTY4vUeBrp+",
+  "key": {
+    "t": "ck",
+    "sym": "LL85V5L0n5024C6XLGe+/mmg9F/VkUbRQycnN28S5L4=",
+    "v": 1
+  }
+}
```
Comment (wmontwe) [inline crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/symCommonDecrypt.json:11]: Linebreak on file end

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/generateSymKey.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    "cipher": "AES",
+    "blockMode": "GCM",
+    "keySize": 256,
+    "key": {
+      "t": "ck",
+      "sym": "LL85V5L0n5024C6XLGe+/mmg9F/VkUbRQycnN28S5L4=",
+      "v": 1
+    }
+  }
+}
```
Comment (wmontwe) [inline crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/generateSymKey.json:13]: Linebreak on file end

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/generateAsymKeys.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    }
+  }
+}
```
Comment (wmontwe) [inline crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/generateAsymKeys.json:21]: Linebreak on file end

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/asymPublicExchangeKey.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+    "t": "apub",
+    "pub": "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA43uqiWS2xJyNjRT5XUJfyIB8Be0LGQYXKrmgKF77DxohrQz3K1fN+l0AdTZeT7u04f5V8BwrpVG5iRDQxKg8JSWghfjs4YqP8JOrQmheQKbrrsTon2PrAStBsSNoyQlngXex88/lgJfRHx0F+mCDnx9Iz8xdHeeleagKe4kPXEIcKCwL6Ib8sMCSASNqPQLReDML42r0HDzqXDqIVZHXoLjmue+oypk1YpvlWeyU9vXJNe2RKWyscLXGxBIUtRC2XHWAZ3QbebRUhQGbMnhTWYvdXliLhxdNvZTNt+HB1iSpvSLv0aOK3WoebsHIhpzsOAn5ENpDGhNANdUmCTEf1wIDAQAB",
+    "v": 1
+}
```
Comment (wmontwe) [inline crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/asymPublicExchangeKey.json:5]: Linebreak on file end

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/asymPrivateExchangeKey.json [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+    "v": 1
+}
```
Comment (wmontwe) [inline crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/asymPrivateExchangeKey.json:5]: Linebreak on file end

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/asymEncrypt.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    "v": 1
+  }
+}
```
Comment (wmontwe) [inline crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/asymEncrypt.json:15]: Linebreak on file end

**Review by wmontwe — CHANGES_REQUESTED** [review]
Is the consent document needed? There are also issues with line breaks on some files and the test config should not be commited

File: crypto/src-gen/commonTest/kotlin/care/data4life/sdk/crypto/test/config/TestConfig.kt (hunk 1 above)
Comment (bitPogo) [reply crypto/src-gen/commonTest/kotlin/care/data4life/sdk/crypto/test/config/TestConfig.kt]: Done

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/symCommonEncrypt.json (hunk 2 above)
Comment (bitPogo) [reply crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/symCommonEncrypt.json]: Done

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/symCommonDecrypt.json (hunk 3 above)
Comment (bitPogo) [reply crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/symCommonDecrypt.json]: Done

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/generateSymKey.json (hunk 4 above)
Comment (bitPogo) [reply crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/generateSymKey.json]: Done

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/asymEncrypt.json (hunk 8 above)
Comment (bitPogo) [reply crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/asymEncrypt.json]: Done

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/asymPrivateExchangeKey.json (hunk 7 above)
Comment (bitPogo) [reply crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/asymPrivateExchangeKey.json]: Done

File: crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/generateAsymKeys.json (hunk 5 above)
Comment (bitPogo) [reply crypto/src/commonTest/resources/design-documents/crypto/test-fixture/v1/generateAsymKeys.json]: Done

**Review by wmontwe — APPROVED** [review]
👍

**Pull request merged**